internal/charmstore/store.go

```diff
@@ -227,6 +227,17 @@ func (p *Pool) requestStoreNB(always bool) (*Store, error) {
 	return store, nil
 }
 
+// newBlobstore creates a new blobstore based on config and the given StoreDatabase.
+func (p *Pool) newBlobstore(db *mgo.Database) *blobstore.Store {
+	// If no BlobStoreProviders are in the config, use the legacy config of
```

Comment:
I'd prefer it if this was done at a different level. I think it might be better if the default provider was added when the config was read.